server/src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -123,22 +125,42 @@ public HypervisorCheckInResult hypervisorUpdate(
 
         Owner owner = this.getOwner(ownerKey);
 
+        if (hostGuestMap.remove("") != null) {
+            log.warn("Ignoring empty hypervisor id");
+        }
+
         // Maps virt hypervisor ID to registered consumer for that hypervisor, should one exist:
         VirtConsumerMap hypervisorConsumersMap =
                 consumerCurator.getHostConsumersMap(owner, hostGuestMap.keySet());
 
+        int emptyGuestIdCount = 0;
+        int emptyHypervisorIdCount = 0;
         Set<String> allGuestIds = new HashSet<String>();
-        for (Entry<String, List<GuestId>> hostEntry : hostGuestMap.entrySet()) {
-            for (GuestId gid : hostEntry.getValue()) {
-                allGuestIds.add(gid.getGuestId());
+        Iterator<Entry<String, ArrayList<GuestId>>> hostEntries = hostGuestMap.entrySet().iterator();
+        while (hostEntries.hasNext()) {
```

Comment:
I would prefer to see these as for loops.  I find that to be more idiomatic to Java.

```java
for (Iterator<Entry<String, ArrayList<GuestId>>> it = hostGuestMap.entrySet().iterator(); it.hasNext(); ) {

}
```

Also I didn't notice anywhere that you were actually using the Entry object's name so why not just do `hostGuestMap.values()`?